src/Entity/User.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Entity;
+
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Ramsey\Uuid\UuidInterface;
+
+/**
+ * @ORM\Table(name="app_user")
+ * @ORM\Entity(repositoryClass="App\Repository\User\UserRepository")
+ */
+class User implements UserInterface
+{
+    /**
+     * @ORM\Id()
+     * @ORM\Column(type="guid", unique=true)
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string", length=100, unique=true)
+     */
+    private $email;
+
+    /**
+     * @ORM\Column(type="string", length=50)
+     */
+    private $username;
+
+    /**
+     * @ORM\Column(type="json")
+     */
+    private $roles = [];
+
+    /**
+     * @ORM\Column(type="string", length=100)
+     */
+    private $password;
+
+    public function __construct(
+        UuidInterface $id,
+        string $email,
+        string $username
+    ) {
+        $this->id = $id->toString();
+        $this->email = $email;
+        $this->username = $username;
+        $this->roles = $this->getRoles();
+    }
+
+    public function upgradeToAdmin(): self
+    {
+        $roles = $this->roles;
+
+        $roles[] = 'ROLE_ADMIN';
+
+        $this->roles = array_unique($roles);
+
+        return $this;
+    }
+
+    public function getRoles(): array
+    {
+        $roles = $this->roles;
+        // guarantee every user at least has ROLE_USER
+        $roles[] = 'ROLE_USER';
+
+        return array_unique($roles);
+    }
+
+    public function getPassword(): string
+    {
+        return (string) $this->password;
+    }
+
+    public function setPassword(string $password): self
+    {
+        $this->password = $password;
+
+        return $this;
```

Comment:
Do not use a fluent interface here, only use them whenever you are building a builder.